lib/buyers/forms.py

```diff
@@ -12,10 +12,10 @@ def clean_pin(pin):
         return pin
 
     if not len(pin) == 4:
-        raise forms.ValidationError(PIN_4_NUMBERS_LONG)
+        raise forms.ValidationError(PIN_4_NUMBERS_LONG, code='invalid')
 
     if not pin.isdigit():
-        raise forms.ValidationError(PIN_ONLY_NUMBERS)
+        raise forms.ValidationError(PIN_ONLY_NUMBERS, code='invalid')
```

Comment:
We could, but I think that diverges from the Django standard where by the code is a generic term not specific to the field:

https://github.com/django/django/search?utf8=%E2%9C%93&q=default_error_messages&type=Code

You've got the code and the field key to pull things back together. If we go for this standard, it could be possible that people would look at the required field and assume it should be like that. Then I have start overriding required, as pin_required.